### PR TITLE
Improvements to auth flow robustness

### PR DESCRIPTION
**Pull request by kevklam**

### CODE REVIEW

**Comment by kevklam** [comment]
#107 #108

**Comment by rgregg** [comment]
@rgregg is added to the review. #Closed

File: OneDriveSDK/Auth/ODBaseAuthProvider.m [hunk 1]
```diff
@@ -94,27 +100,34 @@ - (void)authorizationFlowCompletedWithURL:(NSURL *)endURL
...
         }
     }
     else{
-        completionHandler(error);
+        [self dismissAndCompleteAuthorizationWithError:error completion:completionHandler];
     }
 }
 
+- (void)dismissAndCompleteAuthorizationWithError:(NSError*)error completion:(AuthCompletion)completionHandler {
+    __block ODAuthenticationViewController *authVC = self.authViewController;
```
Comment (rgregg) [inline OneDriveSDK/Auth/ODBaseAuthProvider.m]: __block shouldn't be necessary here. You aren't modifying authVC in the block or expecting modifications to carry outside of the block. A copy should be fine.

**Comment by rgregg** [comment]
:shipit:

**Pull request merged**